modules/common/src/main/java/com/ibm/cloud/cloudant/internal/CloudantBaseService.java

```diff
@@ -53,7 +53,9 @@ protected OkHttpClient configureHttpClient() {
         // If we are using a CouchDB session authenticator we need to customize the cookie jar
         customizeAuthenticator(a -> builder
                 .cookieJar(a.getCookieJar()));
-        return builder.build();
+        OkHttpClient client = builder.build();
+        this.setClient(client);
```

Comment:
I think this is probably fine, but do we know if there are any places this is called where it might be unexpected to set the client?

Reply:
AFAIK, currently there is not.

Reply:
We could add an instance equality check in `setClient` to avoid "double setting" if we already have the same client, but doing that is probably slower than just setting it again anyway so I'm ok with this as-is.